src/v2/Apps/Components/AppShell.tsx

```diff
@@ -48,6 +49,17 @@ export const AppShell: React.FC<AppShellProps> = props => {
     document.body.setAttribute("data-test", "AppReady")
   }, [])
 
+  /**
+   * Wait for route to finish rendering before (possibly) switching out the theme.
+   *
+   * When the route changes, the configured theme will change immediately; this
+   * will cause the styles to update out of sync with the page change. Here we
+   * wait for the route to finish rendering before setting the next theme.
+   */
+  const nextTheme = routeConfig.theme ?? "v2"
+  const [theme, setTheme] = useState<"v2" | "v3">(nextTheme)
```

Comment:
@dzucconi - Oh sorry, i was thinking more about mentioning how we're migrating pages and how some have old palette and new and that's why this is needed. 

(feel free to update or not, its not a blocker or that big of a deal, i can just see it being somewhat confusing down the line)

Reply:
😄  Will sneak in an update in my next PR